dashboard/test/ui/features/star_labs/dance/dance_ai_modal.feature

```diff
@@ -0,0 +1,52 @@
+Feature: Dance Party
+
+  Scenario: Dance AI Modal
+    Given I am on "http://studio.code.org/s/allthethings/lessons/37/levels/3"
+    And I wait for the page to fully load
+    And I select age 10 in the age dialog
+    And I drag block "dance_ai" to block "setup"
+    And I click block field "[data-id='setup'] > [data-id='dance_ai'] > .blocklyEditableText"
+
+    # First time choosing emojis.
+    And I wait for 0.5 seconds
+    And I press ".src-dance-ai-dance-ai-modal-module__emojiIcon:nth-of-type(3)" using jQuery
+    And I press ".src-dance-ai-dance-ai-modal-module__emojiIcon:nth-of-type(15)" using jQuery
+    And I press ".src-dance-ai-dance-ai-modal-module__emojiIcon:nth-of-type(19)" using jQuery
+    And I press "#generate-button" using jQuery
+
+    # Wait until results show.
+    And I wait until element "#use-button" is visible
+
+    # Toggle to code and back.
+    And I press "toggle-code-button" using jQuery
+    And I press "toggle-effect-button" using jQuery
```

Comment:
I don't currently know a good way to check the toggle.  The technique we use [here](https://github.com/code-dot-org/code-dot-org/blob/9170305a8de5c7ae3cd4ded5a7345b20e1a13722/dashboard/test/ui/features/step_definitions/steps.rb#L239-L241) doesn't appear to account for the flip, even though we do use `backface-visibility: hidden` [here](https://github.com/code-dot-org/code-dot-org/blob/9170305a8de5c7ae3cd4ded5a7345b20e1a13722/apps/src/dance/ai/dance-ai-modal.module.scss#L330).  This might be a candidate for an Eyes test, which I'm considering adding as a follow-up to this initial test.

That said, will add something for the explanation mode.